cmd/gaugeExecuteTests.go

```diff
@@ -0,0 +1,77 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+type gaugeExecuteTestsUtils interface {
+	FileExists(filename string) (bool, error)
+	SetEnv([]string)
+	RunShell(shell string, command string) error
+}
+
+type gaugeExecuteTestsUtilsBundle struct {
+	*command.Command
+	*piperutils.Files
+}
+
+func newGaugeExecuteTestsUtils() gaugeExecuteTestsUtils {
+	utils := gaugeExecuteTestsUtilsBundle{
+		Command: &command.Command{},
+		Files:   &piperutils.Files{},
+	}
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func gaugeExecuteTests(config gaugeExecuteTestsOptions, telemetryData *telemetry.CustomData, influx *gaugeExecuteTestsInflux) {
+	utils := newGaugeExecuteTestsUtils()
+
+	influx.step_data.fields.gauge = false
+	err := runGaugeExecuteTests(&config, telemetryData, utils)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+	influx.step_data.fields.gauge = true
+}
+
+func runGaugeExecuteTests(config *gaugeExecuteTestsOptions, telemetryData *telemetry.CustomData, utils gaugeExecuteTestsUtils) error {
+	var gaugeScript string
+	if config.InstallCommand != "" {
```

Comment:
currently `InstallCommand` executed in a shell only works on Linux systems. 

We could use native go functionality in combination with running commands.
This will provide a bit less flexibility but I would say that we can accept this.